src/game/level.cpp

```diff
@@ -10,6 +10,185 @@ void Level::load_json(std::vector<std::shared_ptr<Entity>> *entities, const std:
+
+      simdjson::ondemand::object obj(this->json_data.at_pointer(json_pointer).get_object());
+
+      f64 x          = obj["x"];
+      f64 y          = obj["y"];
+      f64 width      = obj["width"];
+      f64 height     = obj["height"];
+      f64 tileset_id = obj["id"];
+
+      simdjson::ondemand::array property_array = this->json_data.at_pointer(json_pointer + "/properties").get_array();
+      u32 prop_array_size                      = property_array.count_elements();
+      std::string script                       = "return";
+
+      for (u32 i = 0; i < prop_array_size; i++) {
+        simdjson::ondemand::object obj(property_array.at_pointer("/" + std::to_string(i)).get_object());
+
+        script = std::string_view(obj["value"]);
+      }
+
+      objects_content.emplace_back(x, y, width, height, tileset_id, script);
+    }
+
+    simdjson::ondemand::array collisions_array = this->json_data.at_pointer(layer_id + "/2/chunks").get_array();
+
+    u32 co_array_size = collisions_array.count_elements();
+
+    for (u32 i = 0; i < co_array_size; i++) {
+      std::string json_pointer = layer_id + "/2/chunks/" + std::to_string(i);
+
+      simdjson::ondemand::object obj(this->json_data.at_pointer(json_pointer).get_object());
+
+      i64 x = obj["x"];
+      i64 y = obj["y"];
+
+      std::vector<u64> values;
+
+      simdjson::ondemand::array arr((obj["data"].get_array()));
+
+      for (u64 value : arr)
+        values.push_back(value);
+
+      collisions_content.emplace_back(x, y, std::move(values));
+    }
+  }
+
+  simdjson::ondemand::array tilesets = this->json_data["tilesets"];
+  u32 tilesets_size                  = tilesets.count_elements();
+
+  for (int i = 0; i < tilesets_size; i++) {
+    std::string json_pointer = "/tilesets/" + std::to_string(i);
+    std::string_view tileset_id;
+
+    simdjson::ondemand::object obj(json_data.at_pointer(json_pointer).get_object());
+    std::string_view tileset = obj["source"];
+
+    tileset_id_jsons.emplace_back(tileset);
+  }
+
+  for (u32 i = 0; i < tileset_id_jsons.size(); i++)
+    std::cout << tileset_id_jsons[i] << std::endl;
+
+  for (u32 i = 0; i < tileset_id_jsons.size(); i++) {
+    std::string tileset             = tileset_id_jsons.at(i);
+    simdjson::padded_string id_json = simdjson::padded_string::load(tileset);
+
+    this->id_json_data = json_parser.iterate(id_json);
+
+    simdjson::ondemand::object id_obj = id_json_data;
+    std::string_view tileset_id       = id_obj["image"];
+
+    tileset_id_list.emplace_back(tileset_id);
+  }
 }
 
-void Level::update() {}
+void Level::update() {
+  for (const auto &arr : background_content) {
+    u32 x_counter = 0;
+    u32 y_counter = 0;
+
+    u32 x = arr.x;
+    u32 y = arr.y;
+
+    for (u32 i = 0; i < arr.background_data.size(); i++) {
+      u32 data = arr.background_data.at(i);
+
+      if (data > 0) {
+        std::string id              = tileset_id_list.at(data - 1);
```

Comment:
you only use this variable once, hence you can just use it directly:
`ent->set_texture_path(tileset_id_list.at(data - 1));`